crates/prover/src/components/memory.rs

```diff
@@ -19,10 +17,14 @@ use stwo_prover::core::pcs::TreeVec;
 use stwo_prover::core::poly::circle::{CanonicCoset, CircleEvaluation};
 use stwo_prover::core::poly::BitReversedOrder;
 
+use crate::adapter::memory::MemoryBoundaries;
```

Comment:
the only object that should be imported from the `adapter` is the ProverInput to keep the interface as small as possible